File: backend/portfolio/models/social.py
from django.db import models
from .base import TimeStampedModel
from .user import User
from .social_type import SocialType

"""
Model Social
"""
class Social(TimeStampedModel):
    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='socials')
    social_type = models.ForeignKey(SocialType, on_delete=models.CASCADE)
    link = models.URLField()

    def __str__(self):
        return f"{self.social_type.label} - {self.user.username}"
    
    class Meta:
        verbose_name = "Social"
        verbose_name_plural = "Socials"
        ordering = ['-created_at']

File: frontend/src/initialStore/social.ts
import { SocialType } from "./utils";

interface Social {
  id: number;
  socialType: number;
  social_type_detail: SocialType
  user: number;
  link: string;
  created_at: string;
  updated_at: string;
}

export default {
  socials: [] as Social[],
};
